internal/v4/auth.go

```diff
@@ -82,6 +72,34 @@ func (h *Handler) authorize(req *http.Request, acl []string) error {
 	return httpbakery.NewDischargeRequiredError(m, cookiePath, verr)
 }
 
+// checkRequest checks for any authorisation tokens in the request either HTTP
```

Comment:
I'm not sure we need to describe all the implementation in this comment.
I'd limit it to describing the purpose of the function and how to use it.

    // checkRequest checks the authorization provided by the
    // given request. On success, it reports whether the request has admin
    // rights; without admin rights it returns information on the authenticated
    // user.

This is slightly more awkward to phrase because Admin isn't part of
the struct.